### PR TITLE
[Perf] Reuse GDN attention buffer to reduce allocation overhead

### PR DESCRIPTION
**Pull request by cwazai**
## Summary
This PR eliminates the per-forward `torch.zeros` allocation in `Qwen3NextGatedDeltaNet` and `Qwen3_5GatedDeltaNet` by introducing a reusable workspace buffer. The kernel semantics remain unchanged (output tensor is still zeroed before the custom op).

## Changes
- Add `_core_attn_out_buf` persistent buffer in `Qwen3NextGatedDeltaNet.__init__`
- Add `_get_zeroed_core_attn_out()` helper with capacity growth strategy (1.5x)
- Replace `torch.zeros(...)` calls in both `Qwen3NextGatedDeltaNet.forward` and `Qwen3_5GatedDeltaNet.forward`

## Performance Impact

###Benchmarks
Hardware: Nvidia H800, Software: CUDA/driver：13.0
Workload: model：Qwen3.5-35B-A3B

[before] 
``` python
============ Serving Benchmark Result ============
Successful requests:                     1000      
Failed requests:                         0         
Maximum request concurrency:             100       
Benchmark duration (s):                  36.73     
Total input tokens:                      1013750   
Total generated tokens:                  100000    
Request throughput (req/s):              27.23     
Output token throughput (tok/s):         2722.89   
Peak output token throughput (tok/s):    6934.00   
Peak concurrent requests:                189.00    
Total token throughput (tok/s):          30326.15  
---------------Time to First Token----------------
Mean TTFT (ms):                          1274.34   
Median TTFT (ms):                        453.68    
P99 TTFT (ms):                           9623.89   
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          24.09     
Median TPOT (ms):                        24.00     
P99 TPOT (ms):                           28.74     
---------------Inter-token Latency----------------
Mean ITL (ms):                           23.84     
Median ITL (ms):                         14.86     
P99 ITL (ms):                            118.96    
----------------End-to-end Latency----------------
Mean E2EL (ms):                          3658.80   
Median E2EL (ms):                        2828.67   
P99 E2EL (ms):                           12062.10  
==================================================
```
[after]

```python
============ Serving Benchmark Result ============
Successful requests:                     1000      
Failed requests:                         0         
Maximum request concurrency:             100       
Benchmark duration (s):                  29.60     
Total input tokens:                      1013750   
Total generated tokens:                  100000    
Request throughput (req/s):              33.78     
Output token throughput (tok/s):         3377.84   
Peak output token throughput (tok/s):    6709.00   
Peak concurrent requests:                181.00    
Total token throughput (tok/s):          37620.68  
---------------Time to First Token----------------
Mean TTFT (ms):                          553.38    
Median TTFT (ms):                        454.27    
P99 TTFT (ms):                           2425.18   
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          24.17     
Median TPOT (ms):                        24.20     
P99 TPOT (ms):                           27.54     
---------------Inter-token Latency----------------
Mean ITL (ms):                           23.93     
Median ITL (ms):                         14.96     
P99 ITL (ms):                            118.02    
----------------End-to-end Latency----------------
Mean E2EL (ms):                          2946.40   
Median E2EL (ms):                        2839.37   
P99 E2EL (ms):                           4890.67   
==================================================
```
####Accuracy/Precision Test Results
| DataSet                 | \[Feature] Score | Baseline Score | 
| -------------------- | ---------------- | -------------- | 
|C-Eval      |       0.9101    |  0.9049          | 
| mm_bench  |     0.9178    |  0.9168          | 

## Notes for Reviewers
- The workspace buffer is registered with `persistent=False` to avoid affecting checkpointing
- Zeroing is still performed (`out.zero_()`) to maintain exact kernel semantics
- Follow-up: Consider making kernel fully overwrite output to allow `torch.empty` (future PR)

### CODE REVIEW

File: vllm/model_executor/models/qwen3_5.py [hunk 1]
```diff
@@ -175,10 +175,9 @@ def forward(
         # ============================================================
         # Part 2: Core Attention (Custom Op)
         # ============================================================
-        # Note: we should not use torch.empty here like other attention backends,
-        # see discussions in https://github.com/vllm-project/vllm/pull/28182
```
Comment (ZJY0516) [inline vllm/model_executor/models/qwen3_5.py:179]: please keep these comments

File: vllm/model_executor/models/qwen3_next.py [hunk 2]
```diff
@@ -596,10 +652,9 @@ def forward(
         # ============================================================
         # Part 2: Core Attention (Custom Op)
         # ============================================================
-        # Note: we should not use torch.empty here like other attention backends,
-        # see discussions in https://github.com/vllm-project/vllm/pull/28182
```
Comment (ZJY0516) [inline vllm/model_executor/models/qwen3_next.py:600]: same

File: vllm/model_executor/models/qwen3_5.py (hunk 1 above)
Comment (cwazai) [reply vllm/model_executor/models/qwen3_5.py:179]: Thanks for pointing that out. I've reverted the comment changes
